src/components/container/OverviewTable/OverviewTable.tsx

```diff
@@ -0,0 +1,146 @@
+    const [secondaryData, setSecondaryData] = useState<OverviewData<SecondaryOverviewDataType>[]>();
+
+    const primaryHeaderBackgroundColor = resolveColor(layoutContext.colorScheme, props.primaryHeaderBackgroundColor ?? { lightMode: '#e2f1ff', darkMode: '#000' });
+    const primaryHeaderTextColor = resolveColor(layoutContext.colorScheme, props.primaryHeaderTextColor ?? { lightMode: '#000', darkMode: '#8bcbff' });
+
+    const goodValueBackgroundColor = resolveColor(layoutContext.colorScheme, props.goodValueBackgroundColor ?? { lightMode: '#c8ffc2', darkMode: '#000' });
+    const goodValueTextColor = resolveColor(layoutContext.colorScheme, props.goodValueTextColor ?? { lightMode: '#000', darkMode: '#96ff85' });
+
+    const notGoodValueBackgroundColor = resolveColor(layoutContext.colorScheme, props.notGoodValueBackgroundColor ?? { lightMode: '#ffedc2', darkMode: '#000' });
+    const notGoodValueTextColor = resolveColor(layoutContext.colorScheme, props.notGoodValueTextColor ?? { lightMode: '#000', darkMode: '#ffc847' });
+
+    const daysToCompute = props.daysToCompute ?? 28;
+    const today = startOfToday();
+    const startDate = add(today, { days: -daysToCompute });
+
+    const getDataProvider = <T extends OverviewDataType>(dataType: T): OverviewDataProvider<T> => {
+        if (isSurveyDataType(dataType.rawDataType)) {
+            return { type: dataType, dataProvider: getSurveyDataProvider(dataType.rawDataType) };
+        }
+        return { type: dataType, dataProvider: getDailyDataTypeDefinition(dataType.rawDataType).dataProvider };
+    };
+
+    const getPrimaryDataProvider = (): OverviewDataProvider<PrimaryOverviewDataType> => {
+        if (props.preview) {
+            return createPreviewDataProvider(props.primaryDataType);
+        }
+        return getDataProvider(props.primaryDataType);
+    };
+
+    const getSecondaryDataProviders = (): OverviewDataProvider<SecondaryOverviewDataType>[] => {
+        if (props.preview) {
+            return props.secondaryDataTypes.map(createPreviewDataProvider);
+        }
+        return props.secondaryDataTypes.map(getDataProvider);
+    };
+
+    useInitializeView(() => {
+        const primaryDataProvider = getPrimaryDataProvider();
+        primaryDataProvider.dataProvider(startDate, today).then(result => {
+            return { type: primaryDataProvider.type, queryResult: result };
+        }).then(setPrimaryData);
+
+        const secondaryDataProviders = getSecondaryDataProviders();
+        Promise.all(secondaryDataProviders.map(secondaryDataProvider => secondaryDataProvider.dataProvider(startDate, today).then(result => {
+            return { type: secondaryDataProvider.type, queryResult: result };
+        }))).then(setSecondaryData);
+    }, [], [props.preview, props.primaryDataType, props.secondaryDataTypes]);
+
+    if (!primaryData || !secondaryData) {
+        return <div className="mdhui-overview-table" ref={props.innerRef}><LoadingIndicator /></div>;
+    }
```

Comment:
I'm having trouble hitting this state in the storybook to make sure that it works.  I think it's because `useInitializeView` doesn't reset `primaryData` and `secondaryData` to null/undefined values when it goes to re-load data?  So the storybook starts with preview data, and then if you switch it to live, you don't see this loading indicator because the "old" data hangs around in widget's state while the calls are pending.

Reply:
Yeah.  This was intentional to avoid having the table flash and resize each time the screen gets foregrounded.

You can test the loading behavior by hitting the "Remount component" button at the top of the Storybook render panel.

![image](https://github.com/user-attachments/assets/85ca0fff-1cc6-4b08-973a-c32960f44e69)

You can use this while in preview mode or after switching to live to see the loading indicator.